modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ParameterModelValidator.java

```diff
@@ -143,6 +158,32 @@ private void validateParameter(ParameterModel parameterModel, MetadataTypeVisito
     }
   }
 
+  private void validateParameterFieldsHaveGetters(ComponentModel model, MetadataType parameterMetadataType) {
+    String componentTypeName = IntrospectionUtils.getComponentModelTypeName(model);
+    Class<?> parameterType = getType(parameterMetadataType);
+    parameterMetadataType.accept(new MetadataTypeVisitor() {
+
+      @Override
+      public void visitObject(ObjectType objectType) {
+        Collection<ObjectFieldType> parameters = objectType.getFields();
+        Set<String> fieldsWithGetters =
+            getFieldsWithGetters(parameterType).stream().map(TypeUtils::getAlias).collect(Collectors.toSet());
+        Set<String> parameterWithoutGetters =
+            parameters.stream().map(f -> f.getKey().getName().getLocalPart())
+                .filter(fieldName -> !fieldsWithGetters.contains(fieldName)).collect(toSet());
+        if (!parameterWithoutGetters.isEmpty()) {
+          throw new IllegalParameterModelDefinitionException(format("%s '%s' has an argument or return type of type '%s' which contains fields (%s) that doesn't have the corresponding getter methods",
```

Comment:
Thinking about it, this message should be something like "there is a field without getter or a getter without field".